From cebc9f737aeb891c1ef2fd1befe4ad69513f50a8 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 22 Feb 2020 19:20:17 +0200
Subject: [PATCH 090/140] add Pixel accent overlay

	new file:   packages/overlays/AccentColorPixelOverlay/Android.mk
	new file:   packages/overlays/AccentColorPixelOverlay/AndroidManifest.xml
	new file:   packages/overlays/AccentColorPixelOverlay/res/values/colors_device_defaults.xml
	new file:   packages/overlays/AccentColorPixelOverlay/res/values/strings.xml
	modified:   packages/overlays/Android.mk
---
 .../AccentColorPixelOverlay/Android.mk        | 31 +++++++++++++++++++
 .../AndroidManifest.xml                       | 25 +++++++++++++++
 .../res/values/colors_device_defaults.xml     | 22 +++++++++++++
 .../res/values/strings.xml                    | 23 ++++++++++++++
 packages/overlays/Android.mk                  |  1 +
 5 files changed, 102 insertions(+)
 create mode 100644 packages/overlays/AccentColorPixelOverlay/Android.mk
 create mode 100644 packages/overlays/AccentColorPixelOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/AccentColorPixelOverlay/res/values/colors_device_defaults.xml
 create mode 100644 packages/overlays/AccentColorPixelOverlay/res/values/strings.xml

diff --git a/packages/overlays/AccentColorPixelOverlay/Android.mk b/packages/overlays/AccentColorPixelOverlay/Android.mk
new file mode 100644
index 00000000000..28fbfde61ba
--- /dev/null
+++ b/packages/overlays/AccentColorPixelOverlay/Android.mk
@@ -0,0 +1,31 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := AccentColorPixel
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := AccentColorPixelOverlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/AccentColorPixelOverlay/AndroidManifest.xml b/packages/overlays/AccentColorPixelOverlay/AndroidManifest.xml
new file mode 100644
index 00000000000..c6011e228aa
--- /dev/null
+++ b/packages/overlays/AccentColorPixelOverlay/AndroidManifest.xml
@@ -0,0 +1,25 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.color.pixel"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="android" android:category="android.theme.customization.accent_color" android:priority="1"/>
+
+    <application android:label="@string/accent_color_pixel_overlay" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/AccentColorPixelOverlay/res/values/colors_device_defaults.xml b/packages/overlays/AccentColorPixelOverlay/res/values/colors_device_defaults.xml
new file mode 100644
index 00000000000..30e4be626b6
--- /dev/null
+++ b/packages/overlays/AccentColorPixelOverlay/res/values/colors_device_defaults.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <color name="accent_device_default_light">#1A73E8</color>
+    <color name="accent_device_default_dark">#8AB4F8</color>
+</resources>
diff --git a/packages/overlays/AccentColorPixelOverlay/res/values/strings.xml b/packages/overlays/AccentColorPixelOverlay/res/values/strings.xml
new file mode 100644
index 00000000000..c0fe31428cc
--- /dev/null
+++ b/packages/overlays/AccentColorPixelOverlay/res/values/strings.xml
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Pixel accent color name application label. [CHAR LIMIT=50] -->
+    <string name="accent_color_pixel_overlay" translatable="false">Pixel</string>
+</resources>
+
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index cb2afcb62a2..6a1718cf7f2 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -23,6 +23,7 @@ LOCAL_REQUIRED_MODULES := \
 	AccentColorOrchidOverlay \
 	AccentColorSpaceOverlay \
 	AccentColorGreenOverlay \
+	AccentColorPixelOverlay \
 	AccentColorPurpleOverlay \
 	DisplayCutoutEmulationCornerOverlay \
 	DisplayCutoutEmulationDoubleOverlay \
-- 
2.25.1

